static/third/bootstrap-typeahead/typeahead.js

```diff
@@ -152,6 +162,43 @@
     return false;
   }
 
+  , getTypeaheadStyle: function(top_pos, left_pos) {
+
+    function has_scrollbar(element) {
+      if (element.clientHeight < element.scrollHeight) {
+        return true;
+      }
+      return false;
+    }
+
+    function return_css_object(top_pos, left_pos) {
+
+      const caret_coordinates = getCaretCoordinates(this.$element[0], this.$element[0].selectionEnd);
+      const container_width = this.$container.outerWidth(true);
+      const input_width = this.$element.outerWidth(true);
+
+      // Top Coordinates
+      caret_coordinates.top += top_pos;
+      if (has_scrollbar(this.$element[0])) {
+        caret_coordinates.top -= this.$element.scrollTop();
+      }
+
+      // Left Coordinates
+      if (caret_coordinates.left + container_width >= input_width) {
+        caret_coordinates.left = input_width - container_width + this.$element.offset().left;
+      } else {
+        caret_coordinates.left = left_pos + caret_coordinates.left;
+      }
+
+      return { top: caret_coordinates.top, left: caret_coordinates.left };
+    }
+
+    if (this.shift) {
+      return return_css_object.apply(this, [top_pos, left_pos]);
+    }
+    return { top: top_pos, left: left_pos };
+  }  
```

Comment:
Did you borrow this code from somewhere?  If so, we should cite where.

Reply:
Nah, I don't think so that I borrowed this code. Took a few references from :- http://jsfiddle.net/dandv/aFPA7, but that's unrelated to the movement of typeahead along with cursor.

Reply:
No, I don't think I borrowed this code. However, I must have used http://jsfiddle.net/dandv/aFPA7 as a reference, but its unrelated to the movement of typeahead along with cursor.